mycroft/listener/__init__.py

```diff
@@ -199,6 +199,8 @@ def send_unknown_intent():
             with self.loop.lock:
                 try:
                     text = self.loop.stt.execute(audio, language=lang)
+                    if text is None:
```

Comment:
should we handle empty strings to? maybe use `if not text` and catch all falsy values

Reply:
makes sense; I think `None` is the default in plugins when nothing sets `self.text`, but its ultimately up to the plugin author what they return. No reason not to fall back if the base STT didn't recognize anything, it can't hurt..

Reply:
On another read, the check right after this will handle an empty string as valid STT, is this part of the API or should that check be reversed to:

```python
if not text:
    send_unknown_intent()
else:
    ...
```

Reply:
if both fallback and main stt return an empty string i think that means silent audio and is a valid transcription no? 
i want fallback to try a new transcript if main return empty string, but if both return empty then i think the old logic is valid

Reply:
re-reading code, i think that change makes sense in the below lines